chromium_src/ios/web/webui/mojo_facade.mm

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "src/ios/web/webui/mojo_facade.mm"
+
+namespace web {
+bool MojoFacade::IsWebUIMessageAllowedForFrame(WKFrameInfo* frame,
+                                               const GURL& origin,
+                                               NSString** prompt) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  CHECK(prompt && *prompt);
+
+  auto name_and_args =
+      GetMessageNameAndArguments(base::SysNSStringToUTF8(*prompt));
+
+  // If the scheme is untrusted
+  if (name_and_args.name == "Mojo.bindInterface" &&
+      origin.scheme() == "chrome-untrusted") {
+    const base::Value::Dict& args = name_and_args.args;
+    const std::string* interface_name = args.FindString("interfaceName");
+    CHECK(interface_name);
+
+    // Check if the requested interface is registered
+    bool can_bind_interface =
+        web_state_->GetInterfaceBinderForMainFrame()->HasUntrustedInterface(
+            origin, *interface_name);
```

Comment:
is `origin` the frame origin or the main frame? Shouldn't we be using `WKFrameInfo.securityOrigin`? On desktop we would normally get the security origin for checks like this, but not sure cc @diracdeltas

Reply:
The `origin` passed in is created from the frame's security origin: https://source.chromium.org/chromium/chromium/src/+/main:ios/web/web_state/ui/crw_wk_ui_handler.mm;l=255

Reply:
ah, right, but then why does this need `WKFrameInfo`? It doesn't appear to be used.

Reply:
Technically we don't, though oddly feels like we should have an interface binder for non-main frame or maybe check that the frame passed in is the main frame before getting the binder for the main frame?

Reply:
@bridiver This code isn't using `WKFrameInfo` atm. We only use the `origin` to determine if it has an untrusted interface bound.

Reply:
@kylehickinson actually despite the name this applies to any frame. @Brandon-T right, my point is why are we passing it in then?

Reply:
I pass it in so that I can check if the untrusted interface is allowed for that origin.

Reply:
Sorry for the confusion. I see what you mean now. I thought you were asking why is the `origin` being passed in. I didn't realize you were talking about the `WKFrameInfo*` being passed in. It's removed now in: https://github.com/brave/brave-core/pull/26402/commits/86be840e64541b3df0e092c80591843cf1a7e19c